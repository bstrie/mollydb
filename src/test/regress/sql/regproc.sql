--
-- regproc
--

/* If objects exist, return oids */

CREATE ROLE regtestrole;

-- without schemaname

SELECT regoper('||/');
SELECT regoperator('+(int4,int4)');
SELECT regproc('now');
SELECT regprocedure('abs(numeric)');
SELECT regclass('mdb_class');
SELECT regtype('int4');

SELECT to_regoper('||/');
SELECT to_regoperator('+(int4,int4)');
SELECT to_regproc('now');
SELECT to_regprocedure('abs(numeric)');
SELECT to_regclass('mdb_class');
SELECT to_regtype('int4');

-- with schemaname

SELECT regoper('mdb_catalog.||/');
SELECT regoperator('mdb_catalog.+(int4,int4)');
SELECT regproc('mdb_catalog.now');
SELECT regprocedure('mdb_catalog.abs(numeric)');
SELECT regclass('mdb_catalog.mdb_class');
SELECT regtype('mdb_catalog.int4');

SELECT to_regoper('mdb_catalog.||/');
SELECT to_regproc('mdb_catalog.now');
SELECT to_regprocedure('mdb_catalog.abs(numeric)');
SELECT to_regclass('mdb_catalog.mdb_class');
SELECT to_regtype('mdb_catalog.int4');

-- schemaname not applicable

SELECT regrole('regtestrole');
SELECT regrole('"regtestrole"');
SELECT regnamespace('mdb_catalog');
SELECT regnamespace('"mdb_catalog"');

SELECT to_regrole('regtestrole');
SELECT to_regrole('"regtestrole"');
SELECT to_regnamespace('mdb_catalog');
SELECT to_regnamespace('"mdb_catalog"');

/* If objects don't exist, raise errors. */

DROP ROLE regtestrole;

-- without schemaname

SELECT regoper('||//');
SELECT regoperator('++(int4,int4)');
SELECT regproc('know');
SELECT regprocedure('absinthe(numeric)');
SELECT regclass('mdb_classes');
SELECT regtype('int3');

-- with schemaname

SELECT regoper('ng_catalog.||/');
SELECT regoperator('ng_catalog.+(int4,int4)');
SELECT regproc('ng_catalog.now');
SELECT regprocedure('ng_catalog.abs(numeric)');
SELECT regclass('ng_catalog.mdb_class');
SELECT regtype('ng_catalog.int4');

-- schemaname not applicable

SELECT regrole('regtestrole');
SELECT regrole('"regtestrole"');
SELECT regrole('Nonexistent');
SELECT regrole('"Nonexistent"');
SELECT regrole('foo.bar');
SELECT regnamespace('Nonexistent');
SELECT regnamespace('"Nonexistent"');
SELECT regnamespace('foo.bar');

/* If objects don't exist, return NULL with no error. */

-- without schemaname

SELECT to_regoper('||//');
SELECT to_regoperator('++(int4,int4)');
SELECT to_regproc('know');
SELECT to_regprocedure('absinthe(numeric)');
SELECT to_regclass('mdb_classes');
SELECT to_regtype('int3');

-- with schemaname

SELECT to_regoper('ng_catalog.||/');
SELECT to_regoperator('ng_catalog.+(int4,int4)');
SELECT to_regproc('ng_catalog.now');
SELECT to_regprocedure('ng_catalog.abs(numeric)');
SELECT to_regclass('ng_catalog.mdb_class');
SELECT to_regtype('ng_catalog.int4');

-- schemaname not applicable

SELECT to_regrole('regtestrole');
SELECT to_regrole('"regtestrole"');
SELECT to_regrole('foo.bar');
SELECT to_regrole('Nonexistent');
SELECT to_regrole('"Nonexistent"');
SELECT to_regrole('foo.bar');
SELECT to_regnamespace('Nonexistent');
SELECT to_regnamespace('"Nonexistent"');
SELECT to_regnamespace('foo.bar');
