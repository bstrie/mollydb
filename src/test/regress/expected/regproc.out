--
-- regproc
--
/* If objects exist, return oids */
CREATE ROLE regtestrole;
-- without schemaname
SELECT regoper('||/');
 regoper 
---------
 ||/
(1 row)

SELECT regoperator('+(int4,int4)');
    regoperator     
--------------------
 +(integer,integer)
(1 row)

SELECT regproc('now');
 regproc 
---------
 now
(1 row)

SELECT regprocedure('abs(numeric)');
 regprocedure 
--------------
 abs(numeric)
(1 row)

SELECT regclass('mdb_class');
 regclass 
----------
 mdb_class
(1 row)

SELECT regtype('int4');
 regtype 
---------
 integer
(1 row)

SELECT to_regoper('||/');
 to_regoper 
------------
 ||/
(1 row)

SELECT to_regoperator('+(int4,int4)');
   to_regoperator   
--------------------
 +(integer,integer)
(1 row)

SELECT to_regproc('now');
 to_regproc 
------------
 now
(1 row)

SELECT to_regprocedure('abs(numeric)');
 to_regprocedure 
-----------------
 abs(numeric)
(1 row)

SELECT to_regclass('mdb_class');
 to_regclass 
-------------
 mdb_class
(1 row)

SELECT to_regtype('int4');
 to_regtype 
------------
 integer
(1 row)

-- with schemaname
SELECT regoper('mdb_catalog.||/');
 regoper 
---------
 ||/
(1 row)

SELECT regoperator('mdb_catalog.+(int4,int4)');
    regoperator     
--------------------
 +(integer,integer)
(1 row)

SELECT regproc('mdb_catalog.now');
 regproc 
---------
 now
(1 row)

SELECT regprocedure('mdb_catalog.abs(numeric)');
 regprocedure 
--------------
 abs(numeric)
(1 row)

SELECT regclass('mdb_catalog.mdb_class');
 regclass 
----------
 mdb_class
(1 row)

SELECT regtype('mdb_catalog.int4');
 regtype 
---------
 integer
(1 row)

SELECT to_regoper('mdb_catalog.||/');
 to_regoper 
------------
 ||/
(1 row)

SELECT to_regproc('mdb_catalog.now');
 to_regproc 
------------
 now
(1 row)

SELECT to_regprocedure('mdb_catalog.abs(numeric)');
 to_regprocedure 
-----------------
 abs(numeric)
(1 row)

SELECT to_regclass('mdb_catalog.mdb_class');
 to_regclass 
-------------
 mdb_class
(1 row)

SELECT to_regtype('mdb_catalog.int4');
 to_regtype 
------------
 integer
(1 row)

-- schemaname not applicable
SELECT regrole('regtestrole');
   regrole   
-------------
 regtestrole
(1 row)

SELECT regrole('"regtestrole"');
   regrole   
-------------
 regtestrole
(1 row)

SELECT regnamespace('mdb_catalog');
 regnamespace 
--------------
 mdb_catalog
(1 row)

SELECT regnamespace('"mdb_catalog"');
 regnamespace 
--------------
 mdb_catalog
(1 row)

SELECT to_regrole('regtestrole');
 to_regrole  
-------------
 regtestrole
(1 row)

SELECT to_regrole('"regtestrole"');
 to_regrole  
-------------
 regtestrole
(1 row)

SELECT to_regnamespace('mdb_catalog');
 to_regnamespace 
-----------------
 mdb_catalog
(1 row)

SELECT to_regnamespace('"mdb_catalog"');
 to_regnamespace 
-----------------
 mdb_catalog
(1 row)

/* If objects don't exist, raise errors. */
DROP ROLE regtestrole;
-- without schemaname
SELECT regoper('||//');
ERROR:  operator does not exist: ||//
LINE 1: SELECT regoper('||//');
                       ^
SELECT regoperator('++(int4,int4)');
ERROR:  operator does not exist: ++(int4,int4)
LINE 1: SELECT regoperator('++(int4,int4)');
                           ^
SELECT regproc('know');
ERROR:  function "know" does not exist
LINE 1: SELECT regproc('know');
                       ^
SELECT regprocedure('absinthe(numeric)');
ERROR:  function "absinthe(numeric)" does not exist
LINE 1: SELECT regprocedure('absinthe(numeric)');
                            ^
SELECT regclass('mdb_classes');
ERROR:  relation "mdb_classes" does not exist
LINE 1: SELECT regclass('mdb_classes');
                        ^
SELECT regtype('int3');
ERROR:  type "int3" does not exist
LINE 1: SELECT regtype('int3');
                       ^
-- with schemaname
SELECT regoper('ng_catalog.||/');
ERROR:  schema "ng_catalog" does not exist
LINE 1: SELECT regoper('ng_catalog.||/');
                       ^
SELECT regoperator('ng_catalog.+(int4,int4)');
ERROR:  operator does not exist: ng_catalog.+(int4,int4)
LINE 1: SELECT regoperator('ng_catalog.+(int4,int4)');
                           ^
SELECT regproc('ng_catalog.now');
ERROR:  schema "ng_catalog" does not exist
LINE 1: SELECT regproc('ng_catalog.now');
                       ^
SELECT regprocedure('ng_catalog.abs(numeric)');
ERROR:  schema "ng_catalog" does not exist
LINE 1: SELECT regprocedure('ng_catalog.abs(numeric)');
                            ^
SELECT regclass('ng_catalog.mdb_class');
ERROR:  schema "ng_catalog" does not exist
LINE 1: SELECT regclass('ng_catalog.mdb_class');
                        ^
SELECT regtype('ng_catalog.int4');
ERROR:  schema "ng_catalog" does not exist
LINE 1: SELECT regtype('ng_catalog.int4');
                       ^
-- schemaname not applicable
SELECT regrole('regtestrole');
ERROR:  role "regtestrole" does not exist
LINE 1: SELECT regrole('regtestrole');
                       ^
SELECT regrole('"regtestrole"');
ERROR:  role "regtestrole" does not exist
LINE 1: SELECT regrole('"regtestrole"');
                       ^
SELECT regrole('Nonexistent');
ERROR:  role "nonexistent" does not exist
LINE 1: SELECT regrole('Nonexistent');
                       ^
SELECT regrole('"Nonexistent"');
ERROR:  role "Nonexistent" does not exist
LINE 1: SELECT regrole('"Nonexistent"');
                       ^
SELECT regrole('foo.bar');
ERROR:  invalid name syntax
LINE 1: SELECT regrole('foo.bar');
                       ^
SELECT regnamespace('Nonexistent');
ERROR:  schema "nonexistent" does not exist
LINE 1: SELECT regnamespace('Nonexistent');
                            ^
SELECT regnamespace('"Nonexistent"');
ERROR:  schema "Nonexistent" does not exist
LINE 1: SELECT regnamespace('"Nonexistent"');
                            ^
SELECT regnamespace('foo.bar');
ERROR:  invalid name syntax
LINE 1: SELECT regnamespace('foo.bar');
                            ^
/* If objects don't exist, return NULL with no error. */
-- without schemaname
SELECT to_regoper('||//');
 to_regoper 
------------
 
(1 row)

SELECT to_regoperator('++(int4,int4)');
 to_regoperator 
----------------
 
(1 row)

SELECT to_regproc('know');
 to_regproc 
------------
 
(1 row)

SELECT to_regprocedure('absinthe(numeric)');
 to_regprocedure 
-----------------
 
(1 row)

SELECT to_regclass('mdb_classes');
 to_regclass 
-------------
 
(1 row)

SELECT to_regtype('int3');
 to_regtype 
------------
 
(1 row)

-- with schemaname
SELECT to_regoper('ng_catalog.||/');
 to_regoper 
------------
 
(1 row)

SELECT to_regoperator('ng_catalog.+(int4,int4)');
 to_regoperator 
----------------
 
(1 row)

SELECT to_regproc('ng_catalog.now');
 to_regproc 
------------
 
(1 row)

SELECT to_regprocedure('ng_catalog.abs(numeric)');
 to_regprocedure 
-----------------
 
(1 row)

SELECT to_regclass('ng_catalog.mdb_class');
 to_regclass 
-------------
 
(1 row)

SELECT to_regtype('ng_catalog.int4');
 to_regtype 
------------
 
(1 row)

-- schemaname not applicable
SELECT to_regrole('regtestrole');
 to_regrole 
------------
 
(1 row)

SELECT to_regrole('"regtestrole"');
 to_regrole 
------------
 
(1 row)

SELECT to_regrole('foo.bar');
ERROR:  invalid name syntax
SELECT to_regrole('Nonexistent');
 to_regrole 
------------
 
(1 row)

SELECT to_regrole('"Nonexistent"');
 to_regrole 
------------
 
(1 row)

SELECT to_regrole('foo.bar');
ERROR:  invalid name syntax
SELECT to_regnamespace('Nonexistent');
 to_regnamespace 
-----------------
 
(1 row)

SELECT to_regnamespace('"Nonexistent"');
 to_regnamespace 
-----------------
 
(1 row)

SELECT to_regnamespace('foo.bar');
ERROR:  invalid name syntax
